packages/gatsby/src/utils/gatsby-dependents.js

```diff
@@ -1,13 +1,14 @@
 import { store } from "../redux"
+import { memoize } from "lodash"
 import rpt from "read-package-tree"
 
 // Returns [Object] with name and path
-module.exports = async () => {
+module.exports = memoize(async () => {
```

Comment:
I wonder if we need to memoize and not just store the promise... Not sure if there is any benfit over my approach as we use lodash already in gatsby.

```js
let memoizedPromise;
module.exports = () => {
  if (memoizedPromise) {
    return memoizedPromise;
  }

  memoizedPromise = new Promise(async resolve => {
    const { program } = store.getState()
    const allNodeModules = await rpt(program.directory)
    resolve(allNodeModules.children.filter(
      node =>
        (node.package.dependencies && node.package.dependencies[`gatsby`]) ||
        (node.package.peerDependencies && node.package.peerDependencies[`gatsby`])
    ))
  });

  return memoizedPromise;
})

Reply:
Think this is fine since that's what lodash is doing internally in this case anyway